Lib/warnings.py

```diff
@@ -55,11 +54,18 @@ def _formatwarnmsg_impl(msg):
     if msg.source is not None:
         try:
             import tracemalloc
-            tb = tracemalloc.get_object_traceback(msg.source)
-        except Exception:
-            # When a warning is logged during Python shutdown, tracemalloc
-            # and the import machinery don't work anymore
+        except ImportError:
```

Comment:
I think the CI is failing because other exceptions can be propagated from `import tracemalloc` (if you cange this `except ImportError` to `except Exception` the CI passes).

Reply:
"except Exception:" is already used multiple times in the function, so yeah, I'm ok to replace "except ImportError: "with "except Exception:". I added a comment.